### DISCUSSION TITLE
[UTO Logging] Change Unidirectional Topic Operator Logging Level to Debug

### DISCUSSION THREAD

**Discussion by guangyu-yang-rokt** [strimzi/strimzi-kafka-operator#9979] [Q&A] [answered]
Background:

- strimzi version: 0.39.0
- kafka version: 3.5.0
- infra: AWS EKS
- deployment: helm chart
Context:
We are upgrading from strimzi 0.36.0 to 0.39.0. As per https://github.com/strimzi/strimzi-kafka-operator/issues/9465 mentioned, we lose the observability of topic reconciliation process from `topicOperator` Logging. Therefore, we need to change the strimzi topic operator logging level to debug.
here is my attempt:
<img width="611" alt="image" src="https://github.com/strimzi/strimzi-kafka-operator/assets/156621119/44061fb5-cc2e-4deb-a90c-aea69d481045">

I can also see the logging configuration in helm manifest but we are not getting debug level logging from `topic-operator` container even if I tried change topic configs like increasing partitions numbers of `KafkaTopic` CR during the time. The logging shows exactly same info as issue #9465 

Maybe I have missed something here. Thanks in advance for any help!
**Edit**:
After changing the rootLogger.level to DEBUG I got the info I want. But the only valuable logs come from `TopicOperatorEventHandler` 
Is there any way that I can only enable DEBUG level for specified class like `TopicOperatorEventHandler`?

**Comment by scholzj** [comment]
Maybe you should upgrade to 0.40 where the issue you linked to is fixed?

> I can also see the logging configuration in helm manifest 

We provide Helm only for installing the Cluster Operator. So the only logging you can change in the Helm Chart is the logging in the Cluster Operator.

**Reply by guangyu-yang-rokt** [reply]
Thanks for answering! We can upgrade to 0.40 but that will require separate upgrade on kafka. We own 6 clusters on different AWS regions. And for better stability, we only upgrade one region at the time then we can monitor closely if the upgrade cause some issues in production environment. So during the upgrade period there will be some clusters that have to run in 0.39 for sometime, and we don't want to lose the visibility of topic operator in production. Currently, we just go with the `rootLogger.level: DEBUG` but just want to check if there is anyway that can make it less noisy.

**Reply by scholzj** [reply, marked as answer]
Well, you should be able to configure the loggers on a per-class / per-package level to tune them. But I do not have any example to share.
